js_modules/dagster-ui/packages/ui-core/src/assets/asset-checks/AssetChecks.tsx

```diff
@@ -38,93 +63,401 @@ export const AssetChecks = ({
+
+  const executions = queryResult.data?.assetCheckExecutions;
+
+  const runHistory = () => {
+    if (!executions) {
+      return;
+    }
+    return (
+      <div>
+        <Table>
+          <thead>
+            <tr>
+              <th style={{width: '160px'}}>Evaluation Result</th>
+              <th style={{width: '200px'}}>Timestamp</th>
+              <th style={{width: '200px'}}>Target materialization</th>
+              <th>Metadata</th>
+            </tr>
+          </thead>
+          <tbody>
+            {executions.map((execution) => {
+              return (
+                <tr key={execution.id}>
+                  <td>
+                    <AssetCheckStatusTag execution={execution} />
+                  </td>
+                  <td>
+                    {execution.evaluation?.timestamp ? (
+                      <Link
+                        to={linkToRunEvent(
+                          {id: execution.runId},
+                          {stepKey: execution.stepKey, timestamp: execution.timestamp},
+                        )}
+                      >
+                        <TimestampDisplay timestamp={execution.evaluation.timestamp} />
+                      </Link>
+                    ) : (
+                      <TimestampDisplay timestamp={execution.timestamp} />
+                    )}
+                  </td>
+                  <td>
+                    {execution.evaluation?.targetMaterialization ? (
+                      <Link to={`/runs/${execution.evaluation.targetMaterialization.runId}`}>
+                        <TimestampDisplay
+                          timestamp={execution.evaluation.targetMaterialization.timestamp}
+                        />
+                      </Link>
+                    ) : (
+                      ' - '
+                    )}
+                  </td>
+                  <td>
+                    <MetadataCell metadataEntries={execution.evaluation?.metadataEntries} />
+                  </td>
+                </tr>
+              );
+            })}
+          </tbody>
+        </Table>
+        <div style={{paddingBottom: '16px'}}>
+          <CursorHistoryControls {...paginationProps} />
+        </div>
+      </div>
+    );
+  };
+
+  if (!executions) {
+    return (
+      <Box flex={{direction: 'column'}} padding={24}>
+        <Spinner purpose="section" />
+      </Box>
+    );
+  }
+  return <Box flex={{direction: 'column'}}>{runHistory()}</Box>;
+};
+
+const FixedScrollContainer = ({children}: {children: React.ReactNode}) => {
+  // This is kind of hacky but basically the height of the parent of this element is dynamic (its parent has flex grow)
+  // but we don't want it to grow with the content inside of this node, instead we want it only to grow with the content of our sibling node.
+  // This will effectively give us a height of 0
+  const {viewport, containerProps} = useViewport();
+  return (
+    <Box flex={{grow: 1}} {...containerProps} style={{position: 'relative'}}>
```

Comment:
I'm not 100% sure it's relevant, but it sounds like this could be a flexbox setup where the parent has `justify-items: stretch`, pulling all the children to the height of the tallest sibling, and this container box has `min-height: 0`, giving it no intrinsic content size of it's own?